src/averageAges.js

```diff
@@ -55,8 +65,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
-}
+  const children = people.filter(child =>
+    onlyWithSon
+      ? people.find(mother => child.mother === mother.name)
+      && child.sex === 'm'
+      : people.find(mother => child.mother === mother.name)
+  );
+  const ageDifferenceSum = children.reduce((sum, child) =>
+    sum + child.born - people.find(
+      mother => child.mother === mother.name
+    ).born, 0
+  );
+  const averageAgeDiff = getAverageAge(ageDifferenceSum, children.length);
+
+  return averageAgeDiff;
+};
+
+const getAgesSum = (filtratedPeople) =>
+  filtratedPeople.reduce(
+    (sum, year) => (sum + year.died - year.born), 0);
+
+const getAverageAge = (sum, peopleAmount) =>
+  Math.round(sum / peopleAmount * 100) / 100;
```

Comment:
Really amazing work creating callback functions to reduce your code! :D well done!